### PR TITLE
depends: add NM output to gen_id

### PR DESCRIPTION
**Pull request by fanquake**
`NM` is part of the current toolset, and can be set by the user. Include it in `gen_id`.

### CODE REVIEW

File: depends/Makefile [hunk 1]
```diff
@@ -147,8 +147,8 @@ include packages/packages.mk
...
-$(host_arch)_$(host_os)_id:=$(shell env CC='$(host_CC)' C_STANDARD='$(C_STANDARD)' CXX='$(host_CXX)' CXX_STANDARD='$(CXX_STANDARD)' AR='$(host_AR)' RANLIB='$(host_RANLIB)' STRIP='$(host_STRIP)' SHA256SUM='$(build_SHA256SUM)' DEBUG='$(DEBUG)' LTO='$(LTO)' NO_HARDEN='$(NO_HARDEN)' ./gen_id '$(HOST_ID_SALT)' 'GUIX_ENVIRONMENT=$(realpath $(GUIX_ENVIRONMENT))')
+build_id:=$(shell env CC='$(build_CC)' C_STANDARD='$(C_STANDARD)' CXX='$(build_CXX)' CXX_STANDARD='$(CXX_STANDARD)' AR='$(build_AR) 'NM='$(build_NM)' RANLIB='$(build_RANLIB)' STRIP='$(build_STRIP)' SHA256SUM='$(build_SHA256SUM)' DEBUG='$(DEBUG)' LTO='$(LTO)' NO_HARDEN='$(NO_HARDEN)' ./gen_id '$(BUILD_ID_SALT)' 'GUIX_ENVIRONMENT=$(realpath $(GUIX_ENVIRONMENT))')
+$(host_arch)_$(host_os)_id:=$(shell env CC='$(host_CC)' C_STANDARD='$(C_STANDARD)' CXX='$(host_CXX)' CXX_STANDARD='$(CXX_STANDARD)' AR='$(host_AR)' NM='$(build_NM)' RANLIB='$(host_RANLIB)' STRIP='$(host_STRIP)' SHA256SUM='$(build_SHA256SUM)' DEBUG='$(DEBUG)' LTO='$(LTO)' NO_HARDEN='$(NO_HARDEN)' ./gen_id '$(HOST_ID_SALT)' 'GUIX_ENVIRONMENT=$(realpath $(GUIX_ENVIRONMENT))')
```
Comment (TheCharlatan) [inline depends/Makefile:151]: Should this be `host_NM`? This does not rebuild for me if I change the `NM` var other otherwise.

Comment (fanquake) [reply depends/Makefile]: Yep, bad copy-paste.

**Review by sedited — APPROVED** [review]
ACK a1db512f069039530b9fb70f8c004a4f25c3e257

~Guix build (x86)~

**Comment by fanquake** [comment]
@TheCharlatan note that you've ACK'd what I assume was the prior commit hash here

**Comment by TheCharlatan** [comment]
Yeah, also built on the wrong commit, so:
Re-ACK 6ec2813cd88d5f0b955d746e4711a8ad256ea47f

Guix build (x86):
```
67d1d5dd27bc7b2df6b5d77265a9be0c5e3d7cf0bd003b8ec177010c18526814  guix-build-6ec2813cd88d/output/aarch64-linux-gnu/SHA256SUMS.part
59d2f605151251ee05e9c8b043ead799254c0b61da513df8ba4b9adb8c3dd619  guix-build-6ec2813cd88d/output/aarch64-linux-gnu/bitcoin-6ec2813cd88d-aarch64-linux-gnu-debug.tar.gz
d299e058786d3db9e145ea692b4e90b9ce8a60ec16d101e475e8b2927b53f5ac  guix-build-6ec2813cd88d/output/aarch64-linux-gnu/bitcoin-6ec2813cd88d-aarch64-linux-gnu.tar.gz
7732932c2364b7af2369e74e2ff9a0c27373f9e609e1a88f78869637bbccbf60  guix-build-6ec2813cd88d/output/arm-linux-gnueabihf/SHA256SUMS.part
a6bc02461cb6cfe3d96741ec883964fc635dcef5f1ffd47267410984b38ce322  guix-build-6ec2813cd88d/output/arm-linux-gnueabihf/bitcoin-6ec2813cd88d-arm-linux-gnueabihf-debug.tar.gz
4ff7e341fa221bb60de07f8e2e4e45fd4a5fb700c0fd7ce86c7bff49a9b6b37b  guix-build-6ec2813cd88d/output/arm-linux-gnueabihf/bitcoin-6ec2813cd88d-arm-linux-gnueabihf.tar.gz
d78f309152100991c760beac25a71bae13f4a0a4706effc95d09c9470d48ab0e  guix-build-6ec2813cd88d/output/arm64-apple-darwin/SHA256SUMS.part
1386028dff7e8edee95edf11eaf28af6ee628e9aad4a270a67b5c6de8f500318  guix-build-6ec2813cd88d/output/arm64-apple-darwin/bitcoin-6ec2813cd88d-arm64-apple-darwin-unsigned.tar.gz
24e0fe6a1b61487960176da134f7559bbebf9ef868dd3f4a638c703692175ae5  guix-build-6ec2813cd88d/output/arm64-apple-darwin/bitcoin-6ec2813cd88d-arm64-apple-darwin-unsigned.zip
0611092f5ba21b59e246743b6d74c6438ecfe95fd2049d2048459bf59a6896db  guix-build-6ec2813cd88d/output/arm64-apple-darwin/bitcoin-6ec2813cd88d-arm64-apple-darwin.tar.gz
2662fe43573c1c7870bbc59be5b1358a9dc3513664b75f413fe70ca5eba4bbe1  guix-build-6ec2813cd88d/output/dist-archive/bitcoin-6ec2813cd88d.tar.gz
500c979ef50be80750cf85517c86d8db434d388620a5dc4d39f3b3b90afc2224  guix-build-6ec2813cd88d/output/powerpc64-linux-gnu/SHA256SUMS.part
e320f8b7daf2a90fd21bc6f6607cbdfbcada2a045a3fa22acea37e5502d89faf  guix-build-6ec2813cd88d/output/powerpc64-linux-gnu/bitcoin-6ec2813cd88d-powerpc64-linux-gnu-debug.tar.gz
fcdc263f9fb29dd9a468e999968bd799c65ecd04380ce4490647259478ec3cb2  guix-build-6ec2813cd88d/output/powerpc64-linux-gnu/bitcoin-6ec2813cd88d-powerpc64-linux-gnu.tar.gz
a3003fb4916ee2e422d3dec2ec9999fcb60e227554da1c23e9ac3995e8dbafa6  guix-build-6ec2813cd88d/output/powerpc64le-linux-gnu/SHA256SUMS.part
f19c30d6a8e1c3bc29a880ee55fdfc0ee7fdacc1a97c823f00facffdadc1af72  guix-build-6ec2813cd88d/output/powerpc64le-linux-gnu/bitcoin-6ec2813cd88d-powerpc64le-linux-gnu-debug.tar.gz
f38767f7e1f8d0b9accd2e97882dea653564d2c93d657ab8d4bbf15bb150f93d  guix-build-6ec2813cd88d/output/powerpc64le-linux-gnu/bitcoin-6ec2813cd88d-powerpc64le-linux-gnu.tar.gz
6286b59f2810196ff3df8b0c9b542713b7e4c8d15578848f043baf8ccc8f9937  guix-build-6ec2813cd88d/output/riscv64-linux-gnu/SHA256SUMS.part
db222bb8c3df9d89212c3620d8f39780141fe9da8635781dec92a2dacad40ac1  guix-build-6ec2813cd88d/output/riscv64-linux-gnu/bitcoin-6ec2813cd88d-riscv64-linux-gnu-debug.tar.gz
cd9b5fbee646f3b559fad44c3e8f303f5f5850480b92d433ef9601cc057e5c36  guix-build-6ec2813cd88d/output/riscv64-linux-gnu/bitcoin-6ec2813cd88d-riscv64-linux-gnu.tar.gz
291a129b98175d70fa3594032edbfbae24f1a739dc7a066ecc7eeeeaa6076887  guix-build-6ec2813cd88d/output/x86_64-apple-darwin/SHA256SUMS.part
c176b4678eeb33d8800eab9418b6cd262336e72b5aceb41321f51a267dba19cd  guix-build-6ec2813cd88d/output/x86_64-apple-darwin/bitcoin-6ec2813cd88d-x86_64-apple-darwin-unsigned.tar.gz
d5f0b1428338c84782a69c095223b93772cf164c0351ae3b3224f0fe04f93df0  guix-build-6ec2813cd88d/output/x86_64-apple-darwin/bitcoin-6ec2813cd88d-x86_64-apple-darwin-unsigned.zip
6462706848e44c55abdbef86412c6923c3bb7b119139cad91a194d021212f9a0  guix-build-6ec2813cd88d/output/x86_64-apple-darwin/bitcoin-6ec2813cd88d-x86_64-apple-darwin.tar.gz
778c6d1eb7107cd63107305dd6b94573821b080a588e752b3d79c4451630d902  guix-build-6ec2813cd88d/output/x86_64-linux-gnu/SHA256SUMS.part
695dc665c3f03f157e09f8d130c2e172f7817edfed4f7e29c67b203d4ac69121  guix-build-6ec2813cd88d/output/x86_64-linux-gnu/bitcoin-6ec2813cd88d-x86_64-linux-gnu-debug.tar.gz
a04c8f8f5b66989f1d64742358bd52a9df9dc14e90fb24d73dcaaa7612b0d94c  guix-build-6ec2813cd88d/output/x86_64-linux-gnu/bitcoin-6ec2813cd88d-x86_64-linux-gnu.tar.gz
8d74794649c280ca3abd711ded1bfc8ce162f31a5157db543a677d43ab509467  guix-build-6ec2813cd88d/output/x86_64-w64-mingw32/SHA256SUMS.part
62a37721c29c21e007df3edd902a0bae71feec17ab47f8c90ea500f3d9105435  guix-build-6ec2813cd88d/output/x86_64-w64-mingw32/bitcoin-6ec2813cd88d-win64-debug.zip
ec51507636565deecd6f26270e6e8f83232b0b868a27e52e665bfb818ce76b94  guix-build-6ec2813cd88d/output/x86_64-w64-mingw32/bitcoin-6ec2813cd88d-win64-setup-unsigned.exe
b8383c3689da80fd9c7f3dd8b2f9d36be550c1cedd89e7f68f7b4f2357ffecf0  guix-build-6ec2813cd88d/output/x86_64-w64-mingw32/bitcoin-6ec2813cd88d-win64-unsigned.tar.gz
0d5e8d6a5827b318471986eb1de839fd9baf9610ff11a1c50eb29ab59b1bf06f  guix-build-6ec2813cd88d/output/x86_64-w64-mingw32/bitcoin-6ec2813cd88d-win64.zip
```

**Pull request merged**